### PR TITLE
chickenPackages: separate build phases

### PR DESCRIPTION
**Pull request by nagy**
###### Description of changes

The main purpose of this PR is to separate the currently all encompasing installPhase into build- and install- phases.
This makes debugging easier by allowing jumping into a shell with nix-shell.

This also revert a change I made earlier to the wrapper as that has broken some packages.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/chicken/5/eggDerivation.nix [hunk 1]
```diff
@@ -22,6 +22,12 @@ in
 
   CSC_OPTIONS = lib.concatStringsSep " " cscOptions;
 
+  buildPhase = ''
+    runHook preBuild
+    chicken-install -cached -no-install ${lib.concatStringsSep " " chickenInstallFlags}
```
Comment (Et7f3) [inline pkgs/development/compilers/chicken/5/eggDerivation.nix:27]: Add a lib.shellEscape somewhere ?

File: pkgs/development/compilers/chicken/5/eggDerivation.nix [hunk 2]
```diff
@@ -32,15 +38,14 @@ in
     for f in $out/bin/*
     do
       wrapProgram $f \
-        --prefix CHICKEN_REPOSITORY_PATH : "$out/lib/chicken/${toString chicken.binaryVersion}" \
-        --suffix CHICKEN_INCLUDE_PATH : "$out/share" \
+        --prefix CHICKEN_REPOSITORY_PATH : "$out/lib/chicken/${toString chicken.binaryVersion}:$CHICKEN_REPOSITORY_PATH" \
```
Comment (Et7f3) [inline pkgs/development/compilers/chicken/5/eggDerivation.nix:41]: I don't know anything about chicken why this variable was added ?

Comment (nagy) [reply pkgs/development/compilers/chicken/5/eggDerivation.nix:41]: I am just reverting what I did a few PRs back. I dont know why this broke and I have to investigate that, but for now this repairs some programs that come with some eggs.

File: pkgs/development/compilers/chicken/5/eggDerivation.nix (hunk 1 above)
Comment (nagy) [reply pkgs/development/compilers/chicken/5/eggDerivation.nix]: done

**Review by konst-aa — APPROVED** [review]
Looks good to me. TIL what addMetaAttrs exists.

**Pull request merged**